docs/2_1_common_api.adoc

```diff
@@ -213,6 +213,7 @@ The output argument values are not defined, but the computation can continue.
 When debug logging is enabled (`loggingOn == fmi3True`) the function <<logMessage>> was called by the FMU.
 Advanced master implementations can try alternative approaches to drive the simulation by calling the function with different arguments or calling another function.
 Otherwise the master has to treat this return code like <<fmi3Error>> and has to terminate the simulation.
+_[When detecting illegal arguments or the wrong function call at the current state, the FMU may return <<fmi3Discard>> instead of <<fmi3Error>> to allow the importer to attempt an alternative call.]_
```

Comment:
But the master cannot distinguish between a failed doStep due to numerical reasons and a wrong function call e.g. due to illegal arguments. 
Wouldn't it be better to introduce an fmi3Illegal return value clearly telling the master "it's your fault"?

Reply:
Of course there is a large number of cases for errors/issues that we [sc]hould distinguish and not just illegal arguments and numerical errors.
However, the point is not to declare the reasons for failing, but to declare if we can continue and how (or not). And it is a quality of implementation issue if you as master algorithm have a number of heuristics on how to continue when a fmi3Discard comes back.
And if you have no clue, treat fmi3Discard just like fmi3Error.